projects/npm-tools/packages/changelog-generator/src/index.js

```diff
@@ -838,22 +1059,64 @@ async function main(_node, _script, ...args) {
 			.filter(Boolean)
 			.join('\n');
 
-		if (options.dryRun) {
-			process.stdout.write(contents + '\n');
-		}
-		else {
-			await writeFileAsync(outfile, newContents);
-		}
+		await write(options, contents, newContents);
 
 		written = newContents.length;
 	}
 
-	if (options.dryRun) {
+	if (options.interactive && options.version.endsWith(PLACEHOLDER_VERSION)) {
+		info(`[--interactive] Would write ${outfile} ${written} bytes ✨`);
+
+		const answer = await prompt(
+			'Please choose a version from the list, or provide one in full (or enter to abort):',
+			Array.from(DERIVED_VERSIONS)
+		);
+
+		if (answer === '') {
+			process.exit();
+		}
+		else {
+			options.version = answer;
+
+			options.version = await getVersion(options);
+
+			return await go(options);
+		}
+	}
+	else if (options.dryRun) {
 		info(`[--dry-run] Would write ${outfile} ${written} bytes ✨`);
 	}
 	else {
 		info(`Wrote ${outfile} ${written} bytes ✨`);
 	}
+
+	if (options.interactive && !options.dryRun) {
+		info('[--interactive] git-diff preview 👀');
+
+		const diff = await git(
+			'-c',
+			'color.diff=always',
+			'diff',
+			'--',
+			options.outfile
+		);
+
+		log('');
+
+		log(diff);
+
+		const answer = await prompt(
+			'Would you like to stage these changes? (y/n)',
+			[]
+		);
+
+		if (/^y(es?)?$/i.test(answer)) {
+			await git('add', '--', options.outfile);
+		}
+
+		// we appear to hang here... why?
```

Comment:
Whoops. Will blow this away. We were hanging at one point because I wasn't calling `readline.close()`.